src/panels/lovelace/views/hui-view.ts

```diff
@@ -351,6 +424,50 @@ export class HUIView extends LitElement {
       curBadgeEl === badgeElToReplace ? newBadgeEl : curBadgeEl
     );
   }
+
+  private _storeLayout(): void {
```

Comment:
Instead of going over all the card on every drag, can't we just move the card we just dropped?

Reply:
That would be ideal. Just need to find the correct algorithm that finds where the card should be placed and "slices" the card into the config. Something I haven't put much time in yet. Something I need some help with is what to do when saving the lovelace configuration. Saving it causes an update that makes the cards refresh and replace them (same order but cards go from zero height to full height. When I remove the save of the config this issue doesn't exist. Potentially a save placement button?

Reply:
> Potentially a save placement button?
Yeah, just like our card configs, not reason to save anything until after the user is done messing about

Reply:
If this is the route we want to go where would the save button go? If a card is moved replace the + action fab with a Save icon?

Add a second fab above the add card fab that appears if a card is moved? (Material Design says this is okay if the actions are both important. Three is bad)

Add a button to the top of the app header? This would take adding configuration that is passed back to the header that a card was moved.

Whatever we choose I think the best option would not to show this until a card has been moved

Reply:
We should not have a save button, it makes the user experience bad, you should just prevent the update from happening here <https://github.com/home-assistant/home-assistant-polymer/blob/dev/src/panels/lovelace/views/hui-view.ts#L215>.
Btw, what happens when a card changes size due to a state update (conditional) while in edit mode?

Reply:
@bramkragten Checkout my comments about that on this PR 😉